de/tensor.py

```diff
@@ -56,6 +56,19 @@ def form_tensor() -> np.ndarray:
 
     return Tensor, gene_names, cellLines
 
+def factorize(num_comp=6):
+    """ Using Parafac as a tensor factorization. """
+    tensor, genes, cellLines = form_tensor()
+    # perform parafac and CP decomposition
+    r2x_parafac = []
+    for i in range(1, 7):
+        # parafac
+        fac_p = parafac(tensor, rank=i, svd="randomized_svd")
```

Comment:
This should use the same settings as I included in the test. For example, line search helps get a solution faster, and you want the initialization to work beyond 6 components.